api/error.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+(function() {
+
+var
+	that
+;
+
+api.error = that = {
+	// data : [{file, urlError}, ...]
+	throw: function( error, data ) {
+
+		var
+			str
+		;
+
+		switch( error ) {
+			// file can't be played
+			case "INVALID_FORMAT":
+				str = "Error : Invalid file format : The file " + data.file + " can't be played";
+			break;
+			case "URL_UNSUPPORTED":
+				str = "Error : URL not supported : The url " + data.file + " can't be played (" + data.urlError + ")";
+			break;
+			case "URL_UNSUPPORTED_CORS":
+				str = "Error : URL not supported : The url " + data.file + " can't be played (CORS issue)";
+			break;
+			// features not supported by the browser
+			case "FX_OPEN_FILE_SHORTCUT":
+				str = "Warning : Sorry... CTRL+O is disable on Firefox. Use the folder icon (in the lower left corner) or drag and drop to open your files :-)";
```

Comment:
_Warning : Sorry... CTRL+O **isn't working** on your browser.\n
Use the folder icon (in the lower left corner) or drag and drop **your files directly** :-)_